nexus/db-model/src/schema_versions.rs

```diff
@@ -0,0 +1,800 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Database schema versions and upgrades
+//!
+//! For details, see schema/crdb/README.adoc in the root of this repository.
+
+use anyhow::{bail, ensure, Context};
+use camino::Utf8Path;
+use omicron_common::api::external::SemverVersion;
+use once_cell::sync::Lazy;
+use std::collections::BTreeMap;
+
+/// The version of the database schema this particular version of Nexus was
+/// built against
+///
+/// This must be updated when you change the database schema.  Refer to
+/// schema/crdb/README.adoc in the root of this repository for details.
+pub const SCHEMA_VERSION: SemverVersion = SemverVersion::new(46, 0, 0);
+
+/// List of all past database schema versions, in *reverse* order
+///
+/// If you want to change the Omicron database schema, you must update this.
+static KNOWN_VERSIONS: Lazy<Vec<KnownVersion>> = Lazy::new(|| {
+    vec![
+        // +- The next version goes here!  Duplicate this line, uncomment
+        // |  the *second* copy, then update that copy for your version,
+        // |  leaving the first copy as an example for the next person.
+        // v
+        // KnownVersion::new(next_int, "unique-dirname-with-the-sql-files"),
+        KnownVersion::new(46, "first-named-migration"),
```

Comment:
Came up in `#oxide-update` today, but some schema changes will require multiple versions to be correct. (Came up for #5032 earlier, and also for #5287 which requires three migrations.) It would be nice to be able to express an ordered list of versions this way, maybe within a single directory have a bunch of subdirectories.

Reply:
> (Though we think we can simplify this constraint and not require that, by changing the schema version after each file rather than each directory.)

@jgallagher is going to be doing this.

Reply:
Sorry I didn't quite follow the problem.  @sunshowers Is this PR blocked on that work?  Does it make this situation worse?

> Came up in #oxide-update today, but some schema changes will require multiple versions to be correct. (Came up for https://github.com/oxidecomputer/omicron/pull/5032 earlier, and also for https://github.com/oxidecomputer/omicron/pull/5287 which requires three migrations.) It would be nice to be able to express an ordered list of versions this way, maybe within a single directory have a bunch of subdirectories.

For a PR that needs to land more than one separate schema update, is there any reason not to just define two new `KnownVersion` values with separate major numbers?  i.e., why does this use case not fit into the infrastructure we already have / what's added in this PR?